lib/test.js

```diff
@@ -239,31 +233,45 @@ Test.prototype._publicApi = function () {
 		self._assert();
 	}
 
+	function onAssertionEvent(event) {
+		if (event.type === 'error') {
+			event.error.powerAssertContext = event.powerAssertContext;
+			event.error.originalMessage = event.originalMessage;
+			self._setAssertError(event.error);
+			self._assert();
+			return null;
+		}
+
+		var fn = observableToPromise(event.returnValue);
+
+		if (isPromise(fn)) {
+			return Promise.resolve(fn)
+				.catch(function (err) {
+					self._setAssertError(err);
```

Comment:
I am not sure - it might. I need to test that. My initial thought is no, sense the only assertions that return promises is `throws` and `doesNotThrow` right? Those are `non-enhanced`. 

Definitely worth me double checking though.